### PR TITLE
[Merged by Bors] - feat: Define ENorm notation class

### PR DESCRIPTION
**Pull request by fpvandoorn**
* This PR only defines the notation class and two instances, but doesn't use it yet.
* There is an existing unused definition of `ENorm`. Since I might use it in the future, I kept it around, but renamed it (and the file) `ENormedSpace`. I did not modify anything, and didn't add deprecations since it is unused and the main definition now has a different meaning.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

[Zulip thread](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/ENorm.20for.20measure.20theory)

### CODE REVIEW

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

File: Mathlib/Analysis/NormedSpace/ENormedSpace.lean [hunk 1]
```diff
@@ -35,38 +35,42 @@ normed space, extended norm
 noncomputable section
 
 attribute [local instance] Classical.propDecidable
+set_option linter.deprecated false
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/ENormedSpace.lean:38]: @fpvandoorn, why did you add this? I've removed it again in #24001 and everything seems fine.

Comment (fpvandoorn) [reply Mathlib/Analysis/NormedSpace/ENormedSpace.lean:38]: Oh, probably because I was considering deprecating the whole file. Thanks for the fix.
